### PR TITLE
jQuery converted into vanillaJS and jquery.min.js is removed

### PR DESCRIPTION
**Pull request by Askarov5**
####Issue: #50 
I would like to share fully functional only one HTML report file, but because of the (imported) jQuery file it was not fully functional when I share the (master-report)html.

####What does this implement/fix?:
 jQuery code is converted into pure javascript(vanillaJs) and jQuery.min.js file is removed.

### CODE REVIEW

**Comment by rpii** [comment]
Hi, 
thanks for the efforts. 
I am looking at this and testing for regressions.  I have no issue with merging it when its fully functional   I am working through the branch losing the icon on the suite level popout.  Have seen the issue, but no time to debug it for a few days

**Comment by rpii** [comment]
Ok  I merged some changes in to fix the test expansion.  Can you try it with your use case?

**Comment by Askarov5** [comment]
Hi, 
I'm trying to download updated version but I can not see the changes in branches. which branch did you merge?

**Comment by rpii** [comment]
Its in your pull request branch.  its not merged yet

**Comment by Askarov5** [comment]
Then how can I get the updated version to test? I see you also edited/added. Otherwise, if you tested, just merge it.

**Comment by rpii** [comment]
just repull from your repo to where you developed it at.

Rich

File: templates/wdio-html-reporter-template.hbs [hunk 1]
```diff
@@ -392,91 +391,232 @@
...
+        let expandable = document.querySelector('.expandable-control');
+        if (expandable) {
+            expandable.addEventListener('click', event => {
+                event.currentTarget.querySelector('span').classList.toggle('glyphicon-chevron-down');
+                event.currentTarget.querySelector('span').classList.toggle('glyphicon-chevron-up');
+                let panel = event.currentTarget.nextElementSibling.querySelector('.pre.stack.panel');
+                slideToggle(panel, 100);
+            });
+        }
+        if ({{#ifCollapseTests}}{{/ifCollapseTests }}) {
```
Comment (Askarov5) [inline templates/wdio-html-reporter-template.hbs:612]: Tested in safari
The following if block is breaking the code:
```
        if ({{#ifCollapseTests}}{{/ifCollapseTests }}) {
```
Error shown on the console is: SyntaxError: Unexpected token '{'. Expected a property name.
Screen Shot 2021-05-07 at 6 43 45 PM

File: templates/wdio-html-reporter-template.hbs [hunk 2]
```diff
@@ -392,91 +391,232 @@
...
+            event.currentTarget.classList.toggle('filterOff');
         });
-        $(this).toggleClass('filter', 'filterOff', true);
-        $(this).toggleClass('filterOff', 'filter', true);
-    });
 
+        document.querySelector('.togglePassingTests').addEventListener('click', event => {
+            document.querySelectorAll('.suite-pass').forEach(item => {
+                if (item.parentElement.querySelector('span').classList.contains('glyphicon-chevron-up')) {
+                    item.parentElement.querySelector('span').classList.remove('glyphicon-chevron-up');
```
Comment (Askarov5) [inline templates/wdio-html-reporter-template.hbs]: We need to fix togglePassingTests button action(same with the toggleFailingTests): When Test Suite is Slide down(open) if we click on the "HidePassedTests" button, then it's closing only test suite header, but test cases are remaining open and visible.

My solution was: do not chevron-up class from span and just toggle chevron-down class

**Review by Askarov5 — COMMENTED** [review]
Yeah, it worked. There're two issues I can see so far.

**Comment by rpii** [comment]
so the if block is standard handlebars.  Could be you need to define 
            collapseTests: true,  (or false)
in tehereport aggregator ctor.
not sure why that would fail.  I
will look into it, but have no safari.  can you push the changes you want
for toggle tests.  I will pull them and take a look.

**Comment by rpii** [comment]
Ok, made some updates.  Try it out if you can.  would like to merge soon

**Comment by Askarov5** [comment]
Hi Rich, 
Thank you. Everything is working on my side. Tested in chrome, Firefox and safari. Ready for production.

QQ: not related to this topic. I always run tests against multiple browsers and always see the following field empty is this a bug  or configuration issue.
Screen Shot 2021-05-19 at 11 27 06 PM

**Comment by rpii** [comment]
you must set that in your test as it is not available to be queried until
too late in the flow

**Comment by Askarov5** [comment]
I added a question #57 , I would be grateful if you could provide an example there. Thanks.

**Comment by rpii** [comment]
Merging so we can get some better testing

**Pull request merged**